src/app/(nextjs_migration)/(authenticated)/user/layout.tsx

```diff
@@ -0,0 +1,272 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { ReactNode } from "react";
+import Image from "next/image";
+import Script from "next/script";
+
+import "../../../../client/css/index.css";
+import AppConstants from "../../../../appConstants.js";
+import MonitorLogo from "../../../../client/images/monitor-logo-transparent@2x.webp";
+import MozillaLogo from "../../../../client/images/moz-logo-1color-white-rgb-01.svg";
+import { getL10n } from "../../../functions/server/l10n";
+import { getCurrentSession } from "../../../functions/server/sessionHelpers";
+
+export type Props = {
+  children: ReactNode;
+};
+
+const dummyData = {
+  partial: {
+    name: "breaches",
+  },
+  meta: {
+    title: "",
+    socialTitle: "",
+    socialDescription: "",
+  },
+  pathname: "/",
+  fxaProfile: {
+    avatar: "https://profile.accounts.firefox.com/v1/avatar/f",
+  },
+};
+
+import RelayLogo from "../../../../client/images/logo-relay.svg";
+import VPNLogo from "../../../../client/images/logo-vpn.svg";
+import OpenInIcon from "../../../../client/images/icon-open-in.svg";
+import SettingsIcon from "../../../../client/images/icon-settings.svg";
+import HelpIcon from "../../../../client/images/icon-help.svg";
+import SignOutIcon from "../../../../client/images/icon-signout.svg";
+
+const MainLayout = async (props: Props) => {
+  const session = await getCurrentSession();
+  const l10n = getL10n();
+
+  const isBreachesPage = dummyData.partial.name === "breaches";
+  const isSettingsPage = dummyData.partial.name === "settings";
+
+  const UserMenu = ({ userData }) => {
+    const fxaUser = userData?.fxa_profile_json;
+
+    return (
+      <div className="user-menu-wrapper" tabIndex={-1}>
+        <Script src="/nextjs_migration/client/js/userMenu.js" />
+        <button
+          aria-expanded="false"
+          aria-haspopup="true"
+          className="user-menu-button"
+          title={l10n.getString("menu-button-title")}
+        >
+          <img src={fxaUser?.avatar} alt={l10n.getString("menu-button-alt")} />
```

Comment:
Yes I have the same question re: [Image](https://nextjs.org/docs/pages/api-reference/components/image), one of the benefits to switching in the ADR was to get things like the automatic image optimization without having to do the way we do it now.

If however the approach here is to first convert from existing templates -> jsx with a minimum of changes and then later convert to Next components I think that's a fine approach, but we should be explicit if so.

Reply:
FWIW, looks like the only usage of `<img />` tag in src/app/(nextjs_migration/** directory:

```sh
git grep -n "<img " src/app/\(nextjs_migration\) | cat

src/app/(nextjs_migration)/(authenticated)/user/layout.tsx:61:          <img src={fxaUser?.avatar} alt={l10n.getString("menu-button-alt")} />
```

It's been a while since I've looked at Next.js linting, so I might need to read up on https://nextjs.org/docs/pages/building-your-application/configuring/eslint and see what the latest recommendations are.

_"Well, actually&hellip;"_, it looks like running <kbd>npx next lint</kbd> flags this, fascinating!

```sh
npx next lint

./src/app/(nextjs_migration)/(authenticated)/user/layout.tsx
61:11  Warning: Using `<img>` could result in slower LCP and higher bandwidth. Consider using `<Image />` from `next/image` to automatically optimize images. This may incur additional usage or cost from your provider. See: https://nextjs.org/docs/messages/no-img-element  @next/next/no-img-element

...
```

---

**NOTE:** And for anybody else odd enough to try getting into piping and grepping <kbd>next lint</kbd> output, I had to tweak the following to disable colors (since `--no-color` was --no-bueno, and they seem to be writing to STDERR instead of STDOUT):
```sh
FORCE_COLOR=0 npx next lint &> lint.log
```

Took a bit of poking around https://github.com/vercel/next.js/blob/canary/packages/next/src/cli/next-lint.ts and https://github.com/chalk/chalk to piece that together.

Reply:
Thanks for the awesome grep reference!